### PR TITLE
BugFix: useSubscription args watcher creates new subscriptions even when args have not changed

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Uses a better watch method for watching action arguments. Sometimes vue can trigger effects even when the actual value has not changed. This frequently happens when a object is passed as a prop and is then used in a computed argument for a subscription.

### CODE REVIEW

**Review by znicholasbrown — APPROVED** [review]
Overall this makes sense @pleek91, nice work. I had a few non-blocking questions/suggestions

Oh also there's an eslint warning

File: src/utilities/getValidWatchSource.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { WatchSource, WatchEffect, isRef, isReactive } from 'vue'
+import { isArray } from '@/utilities/arrays'
+import { isFunction } from '@/utilities/functions'
+
+export function getValidWatchSource(source: unknown): WatchSource | WatchSource[] | WatchEffect | object {
```
[the remark this answers is not in the document]
Comment (pleek91) [reply src/utilities/getValidWatchSource.ts:5]: These are following vue's definition for `watch`. I tried `Record` but then `reactive()` wasn't working.

[the remark this answers is not in the document]
Comment (znicholasbrown) [reply src/utilities/getValidWatchSource.ts:5]: Hm ok, not keen on it but I get it

File: src/utilities/getValidWatchSource.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  if (isValidWatchValue(source)) {
+    return source
+  }
+
+  if (isArray(source)) {
+    return source.filter(value => isValidWatchValue(value))
```
[the remark this answers is not in the document]
Comment (pleek91) [reply src/utilities/getValidWatchSource.ts:11]: Yep and that is the specific intent of this utility. If you pass an invalid watch source into vue's `watch` function in dev mode you get an ugly warning. The warning makes sense. But here the implementation is for subscriptions  you're passing an array of arguments that may or not be reactive. So this removes anything that's not a valid watch source so you can pass it into a watcher without getting any warnings. 

If something isn't reactive it cannot be watched anyway so there's no change in functionality. This effectively works the same as it did before I just rewrote the utility to be a lot cleaner.

[the remark this answers is not in the document]
Comment (znicholasbrown) [reply src/utilities/getValidWatchSource.ts:11]: Thanks for the explanation 👍🏻

**Review by znicholasbrown — APPROVED** [review]

**Pull request merged**